stdpopsim/genomes.py

```diff
@@ -65,4 +66,8 @@ def recombination_map(self, map_name=None):
         if map_name is None:
             map_name = self.default_genetic_map
         genetic_map = genetic_maps.get_genetic_map(self.species, map_name)
-        return genetic_map.get_chromosome_map(self.name)
+        try:
```

Comment:
OK, I see what you're doing here. We need to be careful about masking other errors here, so I think we should add a method to query whether the chromosome map is defined and do it based on that. This would be more readable as well, and explain what's happening more clearly to the reader.

Reply:
Also, we're not checking to see if it's a chromosome that we know about in the genome. We need to check the chromosome names to see if the are good, then see if there is a map associated with it. If there is no map, then return a flat 0 map. This will just return a flat map for any old key at all, which is definitely a bad idea.